site/profile/manifests/accounts.pp

```diff
@@ -0,0 +1,31 @@
+# This class wraps the puppetlabs/accounts module and the native
+# group type to create accounts and groups from hiera.
+#
+# Enteries in Hiera should look like this:
+#
+# accounts_hash:
+#   bob:
+#     uid: '4001'
+#     gid: '4001'
+#     shell: '/bin/bash'
+#     password: '!!'
+#     sshkeys:
+#       - 'ssh-rsa AAAA...'
+#     locked: false
+#   sue:
+#     uid: '4002'
+#     gid: '4002'
+#     shell: '/bin/ksh'
+#     password: '!!'
+#     sshkeys:
+#       - 'ssh-rsa BBBB...'
+#     locked: false
+class profile::accounts (
+  $accounts = hiera_hash('accounts_hash',{}),
+  $groups = hiera_hash('groups_hash',{}),
+) {
+
+
+  create_resources(accounts::user, $accounts)
```

Comment:
So... looking at writing this with loops.  I'm not quite sure how I make this work with a hash:

$accounts.each |Hash $account| {
  accounts::user { $account:
    ???
  }
}

But will this pass all the parameters correctly?  I'm guessing it wouldn't?  Mind giving me guidance?  Thanks.